dj_search/search.py

```diff
@@ -0,0 +1,143 @@
+from termcolor import colored
+
+
+class DJSearch:
+
+    def __init__(self, db_prefix='', context=None):
+        self.context = context or inspect.currentframe().f_back.f_locals
+        self.virtual_modules = {}
+
+        self.schema_names = [s for s in dj.list_schemas() if s.startswith(db_prefix)]
+
+        tbls_defi = []
+        for schema_name in self.schema_names:
+            self.virtual_modules[schema_name] = dj.create_virtual_module(schema_name, schema_name)
+            schema_defi = self.virtual_modules[schema_name].schema.save()
+            schema_defi = schema_defi.replace('@schema', '@{}'.format(schema_name))
+
+            for match in re.finditer("VirtualModule\('(\w+)', '(\w+)'\)", schema_defi):
+                vmod, vmod_rep = match.groups()
+                schema_defi = schema_defi.replace(vmod, vmod_rep)
+
+            tbls_defi.append(schema_defi)
+
+        # join definitions from all schema, remove INDEX and UNIQUE lines
+        defi = '\n'.join(tbls_defi)
+        defi = re.sub('\s+?INDEX.+?\n|\s+?UNIQUE.+?\n', '', defi, flags=re.MULTILINE)
+        defi = re.sub('([\(\)\[\]\w])( *)"""', '\g<1>\n\g<2>"""', defi)
+
+        self.definition_string = defi
+
+    def search(self, search_str, level=None):
+        """
+        :param search_str: string to search for
+        :param level: 'table', 'attribute', 'comment'
+        :return:
+        """
+        m = DJMatch(search_str, self.definition_string, self.virtual_modules, level=level)
+        m.print()
+        return m
+
+
+class DJMatch:
+
+    def __init__(self, search_str, definition_string, virtual_modules, level=None):
+        self.search_str = search_str
+        self._definition_string = definition_string
+        self._virtual_modules = virtual_modules
+        self.matches = {}
+        self._do_search(level)
+
+    def _do_search(self, level=None):
+        for match in re.finditer('(class\s\w*?)?({})'.format(self.search_str), self._definition_string, re.I):
+            is_class = bool(match.groups()[0])
+
+            if level == 'table':
+                if not is_class:
+                    continue
+                else:
+                    if re.match('(\w+)\(', self._definition_string[match.span(2)[-1]:]):
+                        continue
+
+            # extract the line this matched string belongs to
+            for line_start in re.finditer('\n', self._definition_string[:match.span()[-1]]):
+                pass
+            line_end = re.search('\n', self._definition_string[match.span()[-1]:])
+            line = self._definition_string[line_start.span()[0]:line_end.span()[-1] + match.span()[-1]]
+
+            if ('dj.VirtualModule' in line
+                    or 'dj.Schema' in line
+                    or line.strip() in [f'@{vm}' for vm in self._virtual_modules]):
+                continue
+
+            if is_class:
+                is_attr, is_comment = False, False
+            elif ':' not in line and '#' not in line:
```

Comment:
What if a field comment contains a ":" or a "#"?

Reply:
You could reuse the same parsing structures that DataJoint uses or actually use the table information from the database rather than re-parse the virtual modules.